File: Product_of_Prime_numbers/Primenumbers.py
"""This module helps in checking if a number is prime or not.
Prime numbers are numbers that can only divide 1 and itself without reminder
examples are 2, 3, 5 etc"""
"""How it works
This module accepts a user input and loops through all number from 2 to that input
 then divides the user input by all numbers gotten from the loop.
 it adds this result to a list, if 0 is in the list, user_input is Not prime, if 0 is not
 user input is prime"""


def is_prime_number(user_input):
    """An approach to check if a number is prime or not."""
    return sum([1 if user_input%q == 0 else 0 for q in range(1, user_input+1)]) == 2

File: CHAPTER 1/C PROJECTS/Prime_Numbers.c
/* This program helps in checking if a number is Prime number or not */


#include <stdio.h>

int main(void){
    //At this stage,we are tying to initialize all variables we would be using
    //By doing this, we are statically declaring the data type of all variables to be used
    int number;
    int counter;
    int looper;
    
    counter = 0; 
    printf("Enter the number you want to check: ");
    scanf("%d", &number); //Scanf keyword here, is used to get the user's input
    
    for (looper = 2; looper < number; looper++){
        //This is a for loop in C it takes in 3 parameters start, stop, step 
        //This means that it takes in where to start, where to stop and how much should it jump
        if (number % looper == 0){
            //each number gotten from the loop is divided by the user's input
            //if there is no remainder, counter increases by 1
            counter = counter + 1;
        }
        else{
            //When there is remainder counter increases by 0
            counter = counter + 0;
        }
    }
    if (counter > 0){
        //If the counter increased it means that a number in all the loop was a factor
        //This breaks the law for prime numbers (that they must have no remainder)
        printf("The number is not a Prime Number");
    }
    else{
        printf("The number is a Prime Number");
    }
}

